src/Workleap.Extensions.OpenAPI.Analyzers/CompareTypedResultWithAnnotationAnalyzer.cs

```diff
@@ -168,16 +173,45 @@ public void ValidateEndpointResponseWithAnnotationType(SymbolAnalysisContext con
                 }
             }
 
-            // For the annotations [SwaggerResponse(<StatusCode>, "description", <typeof()>]
+            // For the annotations [SwaggerResponse(<StatusCode>, "description", <typeof()>] and [SwaggerResponse(<StatusCode>, "description", Type = <typeof()>]
             else if (attribute.AttributeClass.ConstructedFrom.Equals(this.SwaggerResponseSymbol, SymbolEqualityComparer.Default))
             {
-                if (attribute.ConstructorArguments.Length > 2 && attribute.ConstructorArguments[0].Value is int statusCodeValue && attribute.ConstructorArguments[2].Value is ITypeSymbol type)
+                if (attribute.NamedArguments.Length > 0 && ValidateTypeInNamedArguments(context, attribute, methodSignatureStatusCodeToTypeMap))
                 {
-                    ValidateAnnotationForTypeMismatch(attribute, statusCodeValue, type, methodSignatureStatusCodeToTypeMap, context);
+                    return;
+                }
+
+                if (attribute.ConstructorArguments.Length > 2 && attribute.ConstructorArguments[0].Value is int statusCodeValue)
+                {
+                    if (attribute.ConstructorArguments[2].Value is ITypeSymbol typeFromAnnotation)
+                    {
+                        ValidateAnnotationForTypeMismatch(attribute, statusCodeValue, typeFromAnnotation, methodSignatureStatusCodeToTypeMap, context);
+                    }
+                    else if (this._statusCodeToResultsMap.TryGetValue(statusCodeValue, out var type))
+                    {
+                        ValidateAnnotationForTypeMismatch(attribute, statusCodeValue, type, methodSignatureStatusCodeToTypeMap, context);
+                    }
                 }
             }
         }
 
+        private static bool ValidateTypeInNamedArguments(SymbolAnalysisContext context, AttributeData attribute,
```

Comment:
For predicate method (returning bools) we often use IsXYZ() to make it more explicit.

Reply:
Also maybe either document or make it more explicit in the name of what exactly do we validate?

Reply:
IsNamedArgumentsTypeMatchMethodSignature

Reply:
Of I see it return a bool AND do the Validate. Could we split in 2 method to have more scoped method? Of even move up the ValidateAnnotationForTypeMismatch in ValidateAnnotationWithTypedResult to keep it more consistant?

Reply:
yup, will just scope this to TryGetTypeFromNamedArguments